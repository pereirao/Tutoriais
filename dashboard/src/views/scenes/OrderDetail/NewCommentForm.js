// @flow

import React, {PureComponent} from 'react';
import {connect} from 'react-redux';
import I18n from '../../../localization';
import {comment_actions} from '../../../business/comment';
import {MBForm, MBFormSubmitButton} from '../../elements/MBForm';
import {FreeResponseField} from '../../shared/form_fields';
import style_sheet from './OrderDetail.module.css';

type NewCommentFormProps = {
  order_id: string,
  isMinibarClosed: () => void,
  is_fetching: boolean, // TODO: do we need this here?
  addComment: (Object) => void
};

export class NewCommentForm extends PureComponent {
  props: NewCommentFormProps

  form: ReactClass<*>

  handleSubmit = data => {
    this.form.reset();
    this.props.addComment(this.props.order_id, data.body);
    this.props.isMinibarClosed();
  };

  render() {
    // const { is_fetching } = this.props;
    return (
      <MBForm
        onSubmit={this.handleSubmit}
        ref={el => {
          this.form = el;
        }}
        className={style_sheet.newCommentForm}>
        <p className={style_sheet.commentDirections}>{I18n.t('ui.body.order_comment_directions')}</p>
        <FreeResponseField name="body"/>
        <MBFormSubmitButton text={I18n.t('ui.button.add_note')}/>
      </MBForm>
    );
  }
}

const NewCommentFormDTP = {addComment: comment_actions.addComment};
const NewCommentFormContainer = connect(null, NewCommentFormDTP)(NewCommentForm);

export default NewCommentFormContainer;
