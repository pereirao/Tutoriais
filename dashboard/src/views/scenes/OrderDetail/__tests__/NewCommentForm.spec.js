import React from 'react';

// NOTE: the named export isn't a connected component
import { NewCommentForm } from '../NewCommentForm';

it('renders correctly', () => {
  expect(shallow(
    <NewCommentForm />
  )).toMatchSnapshot();
});
