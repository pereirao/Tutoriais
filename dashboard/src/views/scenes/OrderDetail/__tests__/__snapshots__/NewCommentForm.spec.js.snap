// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MBForm
  className="newCommentForm"
  field_spacer_width={16}
  onSubmit={[Function]}
  submit_from_last={true}
  validateSubmittable={[Function]}
>
  <p
    className="commentDirections"
  >
    You can type in a custom message. Please be as detailed as possible. Minibar will be notified immediately and will respond here.
  </p>
  <FreeResponseField
    is_required={true}
    name="body"
  />
  <MBFormSubmitButton
    inverted={false}
    is_loading={false}
    text="ADD NOTE"
    title="Submit form"
  />
</MBForm>
`;
